### DISCUSSION TITLE
some wrong on UploadFile methods

### DISCUSSION THREAD

**Discussion by yhfwww** [fastapi/fastapi#7084] [Questions]
####  my code
from fastapi import FastAPI,Form,File,UploadFile
import uvicorn
app = FastAPI()

@app.post("/D2Vuploadfiles/")
async def postlistfiles(files:List[UploadFile]=File(...)):
    for f in files:
        do somethin 
    return {"status":status,"results":ret_list}
if __name__ == "__main__":
    uvicorn.run(app,host='0.0.0.0',port=8000)

####
When I ran the above code, everything was OK. 
And then I go to http://192.168. *. * / docs  to testing the API. I found that when I added two files, the firtst test was successful.
Then I press the remove button after the first file to remove the first file. 
It shows that only the second file is  is still on the page.
However, the API runs for the first file. Is this a bug or is my code wrong?
But when I press the run button again after that, the API runs for the first file. 
Is this a bug or is my code wrong?

**Comment by ycd** [comment]
@yhfwww can you edit your code with proper formatting?

**Comment by krishnardt** [comment]
"ret_list" where is this variable defined?
Is it inside the api or out the apis(globally)?

is it working the same way after restarting the server?

**Comment by tiangolo** [comment]
Hey there, it's not easy to understand the code without formatting and following the new template, so I'll close this issue, but if you still have problems please create a new issue following the template with all the steps and including a self-contained example that shows your use case and error. ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
